### PR TITLE
fix: Checkmarx issue Improper Error Handling

### PR DESCRIPTION
**Pull request by hisarbalik**
## Description

Changes proposed in this pull request (what was done and why):
- Fix Checkmarx issue Improper Error Handling

Changes refer to particular issues, PRs or documents:

- 

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] New features have a milestone set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] Adjusted the documentation if the change is user-facing.
- [x] The feature is unit-tested
- [ ] The feature is e2e-tested

### CODE REVIEW

File: internal/reconciler/metricpipeline/status.go [hunk 1]
```diff
@@ -53,90 +53,65 @@ func (r *Reconciler) updateStatus(ctx context.Context, pipelineName string, lock
...
+	return r.updateGatewayAndAgentStatus(ctx, pipeline)
 
-	if !agentEnabled {
-		return nil
-	}
-	return r.updateAgentStatus(ctx, pipelineName)
 }
 
-func (r *Reconciler) updateGatewayStatus(ctx context.Context, pipelineName string) error {
+func (r *Reconciler) updateGatewayAndAgentStatus(ctx context.Context, pipeline telemetryv1alpha1.MetricPipeline) error {
```
Comment (skhalash) [inline internal/reconciler/metricpipeline/status.go:60]: So what would be the reason if the metric gteway and metric agent are both running?

Comment (hisarbalik) [reply internal/reconciler/metricpipeline/status.go:60]: Reason would be MetricGatewayDeploymentReady in this case

**Review by skhalash — APPROVED** [review]

**Pull request merged**